### PR TITLE
DHFPROD-5912: Initial read capability for Spark connector

### PR DESCRIPTION
**Pull request by rjrudin**
### Description

As covered by the JIRA ticket, this is intentionally constrained to the minimum needed to read Customer instances in the example project. The ReadCustomers program can be used for manual verification of that. 

Moved Customer/ReferenceModelProject into testFixtures in API project so they can be reused in Spark project.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by rjrudin** [comment]
Making this a draft - going to rework the implementation so that the inputs are immediately converted to a plan that is then "enhanced".

File: marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubInputPartitionReader.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+     * Just doing the bare minimum here for now, will have lots more to support soon.
+     *
+     * @param field
+     * @return
+     */
+    private Object readValue(StructField field) {
+        final String fieldName = field.name();
+        Object value;
+        switch (field.dataType().typeName()) {
+            case "integer":
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubInputPartitionReader.java:142]: This is only supporting what's need for the Customer view in the reference-entity-model project. Will be expanded via a separate story.

**Comment by rjrudin** [comment]
Added comments to further explain what's going on.

File: marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java [hunk 2]
```diff
@@ -16,28 +16,51 @@
...
+    @Override
+    public DataSourceReader createReader(DataSourceOptions options) {
+        // Logging of the HubDataSourceReader's constructor indicates that in a simple Spark test program, this method
+        // is called multiple times. On the first occasion, the getSchema method is called, and then instance is seemingly
+        // discarded. On the second occasion, the planInputPartitions method is called, which allows for partition
+        // readers to then be created. And then this is called on a third occasion for unknown reasons. For performance
+        // reasons then, this class will only create one HubDataSourceReader. Testing has shown that if the Spark
+        // program then calls "format" again on a SQLContext, a new instance of this class - DefaultSource - will be
+        // created, thus ensuring that a new data source reader is created.
+        if (hubDataSourceReader == null) {
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java:61]: We'll have to keep an eye on this. Creating multiple HubDataSourceReader objects isn't good, but I'm worried there's some downside to caching this instance that I haven't found yet.

**Review by anu3990 — APPROVED** [review]

**Review by srinathgit — APPROVED** [review]

**Pull request merged**